dashboard/app/models/user.rb

```diff
@@ -2722,6 +2722,20 @@ def personal_account?
     end
   end
 
+  # Determine a student' state based on their teacher's state. Used for student
+  # accounts created prior to the addition of the us_state field in June 2023.
+  # For newer accounts, use the us_state field instead.
+  def get_us_state_from_teacher
```

Comment:
Please try to avoid `get_` and `set_` in method names: https://github.com/rubocop/ruby-style-guide#accessormutator-method-names
```suggestion
  def teacher_us_state
```